File: app/src/main/java/com/factor/launcher/ui/CustomFlag.java
package com.factor.launcher.ui;

import android.content.Context;
import androidx.appcompat.widget.AppCompatTextView;
import com.factor.launcher.R;
import com.skydoves.colorpickerview.AlphaTileView;
import com.skydoves.colorpickerview.ColorEnvelope;
import com.skydoves.colorpickerview.flag.FlagView;

public class CustomFlag extends FlagView {

    private final AppCompatTextView textView;
    private final AlphaTileView alphaTileView;

    public CustomFlag(Context context, int layout) {
        super(context, layout);
        textView = findViewById(R.id.flag_color_code);
        alphaTileView = findViewById(R.id.flag_color_layout);
    }

    @Override
    public void onRefresh(ColorEnvelope colorEnvelope) {
        textView.setText("#" + colorEnvelope.getHexCode());
        alphaTileView.setPaintColor(colorEnvelope.getColor());
    }
}

File: app/src/main/java/com/factor/launcher/ui/CustomFlag.kt
package com.factor.launcher.ui

import android.annotation.SuppressLint
import android.content.Context
import androidx.appcompat.widget.AppCompatTextView
import com.factor.launcher.R
import com.skydoves.colorpickerview.AlphaTileView
import com.skydoves.colorpickerview.ColorEnvelope
import com.skydoves.colorpickerview.flag.FlagView


@SuppressLint("ViewConstructor")
class CustomFlag(context: Context?, layout: Int) : FlagView(context, layout)
{
    private val textView: AppCompatTextView = findViewById(R.id.flag_color_code)
    private val alphaTileView: AlphaTileView = findViewById(R.id.flag_color_layout)

    override fun onRefresh(colorEnvelope: ColorEnvelope)
    {
        textView.text = "#$colorEnvelope.hexCode"
        alphaTileView.setPaintColor(colorEnvelope.color)
    }

}